scheduler.py

```diff
@@ -53,6 +53,15 @@ def neighboors(self, state):
                     d[project] = slot
                     new_state = list(d.items())
                     neighboors.append(new_state)
+
+        for proj1, proj2 in combinations(state, 2):
```

Comment:
For a better undestanding you can do 
`for (proj1, slot1), (proj2, slot2) in combinations(state, 2):`